# Statistical Analysis

### Loading our cleaned data

In [2]:
from load_cleaned_data import load_cleaned_data
import polars as pl
import py_stats as ps

df = load_cleaned_data()

### Descriptive Statistics 

#### Central tendency and dispersion
- **Mean Social Media Usage** - The average time spent online across the entire dataset.
- **Standard Deviation of Stress** - The degree of variation in stress levels among respondents.

In [45]:
def descriptive_analytics(data):
    # Calculate measures of central tendency
    mean = ps.arithmetic_mean(data)
    median = ps.median(data)
    mode = ps.mode(data)
    # Calculate measures of dispersion
    variance = ps.variance(data,population=False)
    std_dev = ps.standard_deviation(data, population=False)
    range = ps.midrange(data)
    context = {
        "mean": mean,
        "median": median,
        "mode": mode,
        "variance": variance,
        "std_dev": std_dev,
        "range": range  
    }

    return context  

In [46]:
# Run descriptive analytics for all numerical columns and display nicely
results = {}
numeric_cols = [col for col, dtype in zip(df.columns, df.dtypes) if dtype in (pl.Int64, pl.Int32, pl.Float64, pl.Float32)]
for col in numeric_cols:
    stats = descriptive_analytics(df[col].to_list())
    results[col] = stats


In [47]:
# Prepare data for DataFrame
stats_rows = []
for col, stats in results.items():
    row = {"column": col}
    row.update(stats)
    stats_rows.append(row)

stats_df = pl.DataFrame(stats_rows)
stats_df

column,mean,median,mode,variance,std_dev,range
str,f64,f64,list[f64],f64,f64,f64
"""age""",15.924722,16.0,[13.0],4.086794,2.021582,16.0
"""daily_social_media_hours""",4.478785,4.4,[3.5],4.072426,2.018025,4.5
"""sleep_hours""",6.494183,6.5,"[6.8, 7.4]",2.050668,1.432015,6.5
"""screen_time_before_sleep""",1.742258,1.8,[1.4],0.511843,0.715432,1.75
"""academic_performance""",2.990248,2.99,"[2.36, 3.9]",0.333279,0.577303,3.0
"""physical_activity""",1.016168,1.0,[1.9],0.33949,0.582658,1.0
"""stress_level""",5.365269,5.0,[4.0],8.363895,2.89204,5.5
"""anxiety_level""",5.557742,6.0,[6.0],8.123589,2.850191,5.5
"""addiction_level""",5.571429,6.0,[6.0],7.988258,2.826351,5.5


# Outcome 
- **Mean Social Media Usage**  = **4.5**
- **Standard Deviation of Stress** = **2.89**

#### Distribution analysis
- **Age Group Distribution** - A breakdown of the population across different age brackets.
- **Gender Representation** - The proportional split of male, female, and other identities in the study.

In [5]:
import plotly.express as px

In [3]:
# lets load the data we want to analyse
age =df['age']

In [ ]:
# create a histogram of the age distribution
fig = (px.histogram(age, x="age", nbins=25, title="Distribution of Age"))

In [37]:
# save the plot
fig.write_html("../visualizations/age_distribution_histogram.html")

In [36]:
# create a boxplot of the age distribution
fig = (px.box(age, y="age", title="Boxplot of Age Distribution"))

In [ ]:
# save the plot
fig.write_html("../visualizations/age_distribution_box_plot.html")

In [40]:
# Gender representaion 
gender = df['gender']

In [ ]:
# histogram representation
fig = (px.histogram(gender, x="gender", title="Distribution of Gender"))

In [ ]:
# save the plot
fig.write_html("../visualizations/gender_representation_distribution_histogram.html")

In [43]:
# box plot presentation
fig = (px.box(gender, y="gender", title="Boxplot of Gender Representation"))

In [44]:
fig.write_html("../visualizations/gender_representation_distribution_box_plot.html")

In [49]:
# distribution based on gender and age
fig = (px.histogram(df, x="age", color="gender", nbins=25, title="Distribution of Age by Gender"))

In [50]:
fig.write_html("../visualizations/Distribution of Age by Gender.html")

# Correlation Analysis

#### Correlation Matrix 
- **Screen Time Before Sleep vs. Sleep Hours** - Analyzes the relationship between late-night device use and rest hours.
- **Social Media Hours vs. Anxiety** - Examines the link between time spent on platforms and anxiety scores.

In [ ]:
# Checking if there is any correlation between screen time before sleep and sleep hours 
screen_time = df['screen_time_before_sleep'].to_list()
sleep_hours = df['sleep_hours'].to_list()


# pearson correlation package imported from py_stats as ps.pearson_correlation
correlation = ps.pearson_correlation(screen_time, sleep_hours)
# 0.0<0.1 = No correlation 0.1<0.3 = little correlation 03<05 medium correlation 05<07 high correlation 0.7 < 1 very high correlation
if correlation == 0.0 or correlation <= 0.1:
    print("No Correlation")
elif correlation == 0.1 or correlation <= 0.3:
    print("Little Correlation")
elif correlation == 0.3 or correlation <= 0.5:
    print("Medium Correlation")
elif correlation == 0.5 or correlation <= 0.7:
    print("High Correlation")
else:
    print("Very High Correlation")
print(f"Pearson Correlation between Screen Time Before Sleep and Sleep Hours: {correlation}")   

No Correlation
Pearson Correlation between Screen Time Before Sleep and Sleep Hours: 0.009925695398454736


In [ ]:
# Checking if there is any correlation between social media hours and anxiety levels    
daily_social_media_hours = df['daily_social_media_hours'].to_list()
anxiety_level = df['anxiety_level'].to_list()

# pearson correlation package imported from py_stats as ps.pearson_correlation
correlation = ps.pearson_correlation(daily_social_media_hours, anxiety_level)

# 0.0<0.1 = No correlation 0.1<0.3 = little correlation 03<05 medium correlation 05<07 high correlation 0.7 < 1 very high correlation
if correlation == 0.0 or correlation <= 0.1:
    print("No Correlation")
elif correlation == 0.1 or correlation <= 0.3:
    print("Little Correlation")
elif correlation == 0.3 or correlation <= 0.5:
    print("Medium Correlation")
elif correlation == 0.5 or correlation <= 0.7:
    print("High Correlation")
else:
    print("Very High Correlation")
print(f"Pearson Correlation between daily socialy media hours and anxiety level: {correlation}")   

Little Correlation
Pearson Correlation between daily socialy media hours and anxiety level: 0.2


#### Visualize with heat maps
- **Mental Health Intensity Map** - A visual matrix showing the intersection of stress, anxiety, and addiction.

In [85]:
# Correlation Intersection between stress level, anxiety level and addiction level
corr_df = df[['stress_level','anxiety_level','addiction_level']]
corr_matrix = corr_df.corr()
corr_matrix

stress_level,anxiety_level,addiction_level
f64,f64,f64
1.0,-0.013311,0.004085
-0.013311,1.0,0.038732
0.004085,0.038732,1.0


In [83]:
fig = px.imshow(corr_matrix,text_auto=True)
fig.write_html("../visualizations/correlation_matrix_stress_anxiety_addiction.html")

### Cohort Analysis 
- **Age-Based Usage Trends** - Comparing digital behavior and mental health outcomes across different age cohorts.

In [62]:
df.schema

Schema([('age', Int64),
        ('gender', String),
        ('daily_social_media_hours', Float64),
        ('platform_usage', String),
        ('sleep_hours', Float64),
        ('screen_time_before_sleep', Float64),
        ('academic_performance', Float64),
        ('physical_activity', Float64),
        ('social_interaction_level', String),
        ('stress_level', Int64),
        ('anxiety_level', Int64),
        ('addiction_level', Int64),
        ('depression_label', Int64)])